library/src/main/java/com/novoda/downloadmanager/MigrationServiceBinder.java

```diff
@@ -29,9 +29,11 @@ public void bind() {
     }
 
     public void unbind() {
-        context.unbindService(serviceConnection);
-        context.stopService(new Intent(context, LiteDownloadMigrationService.class));
-        serviceConnection = null;
+        if (LiteDownloadMigrationService.isBound() && migrationCallback != null) {
```

Comment:
When you bind here, you can hold a reference to the service so that this can be non-static. Also I'm not sure all the static access to the service is needed (regarding `MigrationDownloadServiceBinder`), you can check how is done here https://github.com/novoda/download-manager/blob/epic_download_manager_v2/library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java#L166

Reply:
It seems that we never `unbind` the `DownloadService` can you explain the reasoning? As far as I am aware we should always `unbind` where a corresponding `bind` exists. 

I'll change the above to introduce an interface anyway and use that instead of having the `isBound` etc.